install.sh

```diff
@@ -0,0 +1,229 @@
+#!/bin/sh
+set -eu
+
+# Copyright 2020 The NATS Authors
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# We are sh, not bash; we might want bash/zsh for associative arrays but some
+# OSes are currently on bash3 and removing bash, while we don't want a zsh
+# dependency; so we're sticking to "pretty portable shell" even if it's a
+# little more convoluted as a result.
+#
+# We rely upon the following beyond basic POSIX shell:
+#  1. A  `local`  command (built-in to shell; almost all sh does have this)
+#  2. A  `curl`   command (to download files)
+#  3. An `unzip`  command (to extract content from .zip files)
+#  4. A  `jq`     command (to manipulate JSON)
+#  5. A  `mktemp` command (to stage a downloaded zip-file)
+
+# Like the Python before us, we only support amd64 for now, and that's all that
+# nsc releases.  By calling this out up top, it should make it easier to find
+# the places which matter should that change in the future.
+readonly RELEASE_ARCH="amd64"
+
+# Finding the releases to download
+readonly GITHUB_OWNER_REPO='nats-io/nsc'
+readonly GITHUB_API_NSC="https://api.github.com/repos/$GITHUB_OWNER_REPO"
+readonly HTTP_USER_AGENT='nsc_install/0.1 (@nats-io)'
+readonly GITHUB_HTTP_ACCEPT='Accept: application/vnd.github.v3+json, application/json'
+
+# Where to install to, relative to home-dir
+readonly NSC_RELATIVE_BIN_DIR='.nsccli/bin'
```

Comment:
My best guess is that it's because the Python also runs on Windows, where we don't have such niceties, so picking one directory and sticking with it, cross platform, is the way forward.

I don't think we want to make the install location vary by which interpreter was chosen for the install script, so we absolutely need to match what the Python program does.  We change both or neither.

Sticking to the current location but adding a symlink in ~/bin if ~/bin exists and is in $PATH seems sane.

Reply:
I haven't changed the Python yet.  I'm creating the symlink if `~/bin/` exists, and listing it, then constraining the installation instructions based upon that being in $PATH.

Reply:
Python updated.